libraries/tock-cells/src/optional_cell.rs

```diff
@@ -34,11 +44,25 @@ impl<T: Copy> OptionalCell<T> {
         self.value.set(None);
     }
 
+    /// Return the contained value and replace it with None.
+    pub fn take(&self) -> Option<T> {
+        self.value.take()
+    }
+
     /// Call a closure on the value if the value exists.
     pub fn map<F, R>(&self, closure: F) -> Option<R>
     where
         F: FnOnce(&mut T) -> R,
     {
         self.value.get().map(|mut val| closure(&mut val))
     }
+
+    /// Call a closure on the value if the value exists, or return value
```

Comment:
Are there two different `value`s in this sentence? One is the value inside the cell if it's full, the other is the value passed to the function, right? Can we call one the cell contents or something? I had trouble reading this